andela/coding-geeks-ah-backend
# #167484610 implemented article reading statistics

#### What does this PR do?
It implements the approach of being able to retrieve articles' reading statistics.

#### Description of Task to be done?
1. Adding code to record a specific article's retrieval.
2. Adding a route for retrieving an author's articles' reading statistics.
3. Adding a controller for retrieving an authors' articles' reading statistics.
4. Adding tests for reading statistics.

#### Screenshots
<img width="1384" alt="Screen Shot 2019-09-02 at 21 46 26" src="https://user-images.githubusercontent.com/34879554/64131777-2c7dae00-cdcb-11e9-89c4-926269f9be07.png">

<img width="1133" alt="Screen Shot 2019-09-02 at 21 47 18" src="https://user-images.githubusercontent.com/34879554/64131787-43240500-cdcb-11e9-8557-4982f4fcd82b.png">


#### How should this be manually tested?
1. Clone this branch.
2. Do **npm install**.
4. Do **npm start**
5. Create an article.
6. Check how many times it has been read with the following endpoint: **/api/v1/articles/{articleSlug}/statistics**
7. Retrieve the article that you have created  with the following endpoint: **/api/v1/articles/{articleSlug}**
8. Re-Check how many times it has been read with the following endpoint: **/api/v1/articles/{articleSlug}/statistics** and you will find that it has been read once. Therefore, the more you retrieve it, the more reads increase**
